src/TornCityBot.user.js

```diff
@@ -4,136 +4,231 @@
-                window.location.href = '/gym.php';
-            }
+function checkStats() {
+    var Energy = document.getElementById('energy').children[1].dataset.current;
+    var Nerve = document.getElementById('nerve').children[1].dataset.current;
+    var Happy = document.getElementById('happy').children[1].dataset.current;
+    var Life = document.getElementById('life').children[1].dataset.current;
+    return {
+        'energy': Energy,
+        'nerve': Nerve,
+        'happy': Happy,
+        'life': Life,
+    };
+}
+
+function doGym(step) {
+    if (step === 1) {
+        console.log("doGym step 1");
+        GM_setValue('gym_step', 2);
+        if (window.location.pathname == '/gym.php'){
+            doGym(2);
+        } else {
+            window.location.href = '/gym.php';
         }
-        if (step === 2) {
-            console.log("doGym step 2");
-            if (window.location.pathname != '/gym.php') {
-                window.location.href = '/gym.php';
-            }
-            var Strength = document.getElementById('strengthTotal').innerText;
-            var Defense = document.getElementById('defenceTotal').innerText;
-            var Speed = document.getElementById('speedTotal').innerText;
-            var Dexterity = document.getElementById('dexterityTotal').innerText;
+    }
+    if (step === 2) {
+        console.log("doGym step 2");
+        if (window.location.pathname != '/gym.php') {
+            window.location.href = '/gym.php';
+        }
+        var Strength = document.getElementById('strengthTotal').innerText;
+        var Defense = document.getElementById('defenceTotal').innerText;
+        var Speed = document.getElementById('speedTotal').innerText;
+        var Dexterity = document.getElementById('dexterityTotal').innerText;
 
-            GM_setValue('gym_step', 1);
+        GM_setValue('gym_step', 1);
 
-            if (Strength <= Defense) {
-                document.getElementById('divStrength').getElementsByClassName('numb')[0].value = 1;
-                document.getElementById('divStrength').getElementsByClassName('trainStat')[0].click();
-            } else if (Defense <= Speed) {
-                document.getElementById('divDefence').getElementsByClassName('numb')[0].value = 1;
-                document.getElementById('divDefence').getElementsByClassName('trainStat')[0].click();
-            } else if (Speed <= Dexterity) {
-                document.getElementById('divSpeed').getElementsByClassName('numb')[0].value = 1;
-                document.getElementById('divSpeed').getElementsByClassName('trainStat')[0].click();
+        if (Strength <= Defense) { {setTimeout(function(){
+            document.getElementById('divStrength').getElementsByClassName('numb')[0].value = 20;
+            document.getElementById('divStrength').getElementsByClassName('trainStat')[0].click();
+            }, Math.round(Math.random() * (3000 - 250)) + 250);
+                                   }
```

Comment:
The following lines have closing bracket `}` with wrongful indentation. Please fix this.